From 968928240a632483aea8550d3fad535eb7176625 Mon Sep 17 00:00:00 2001
From: Vladimir 'phcoder' Serbinenko <phcoder@gmail.com>
Date: Sat, 8 Jul 2023 15:50:23 +0200
Subject: [PATCH 37/57] lib/i386/relocator64: Fix 64-bit FreeBSD boot on BIOS

The commit 80948f532d (lib/i386/relocator64: Build fixes for i386) has
broken 64-bit FreeBSD boot on BIOS. This patch fixes the issue.

Fixes: 80948f532d (lib/i386/relocator64: Build fixes for i386)

Signed-off-by: Vladimir Serbinenko <phcoder@gmail.com>
Reviewed-by: Daniel Kiper <daniel.kiper@oracle.com>
---
 grub-core/lib/i386/relocator64.S | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/grub-core/lib/i386/relocator64.S b/grub-core/lib/i386/relocator64.S
index 00bf31599..c80538e7e 100644
--- a/grub-core/lib/i386/relocator64.S
+++ b/grub-core/lib/i386/relocator64.S
@@ -151,7 +151,7 @@ VARIABLE(grub_relocator64_rdx)
 
 #if defined (__APPLE__) || !defined (__x86_64__)
 	.byte 0xff, 0x25
-	.quad 0
+	.long 0
 #else
 	jmp *LOCAL(jump_addr) (%rip)
 #endif
-- 
2.42.0

